src/portal_visualization/builders/imaging_builders.py

```diff
@@ -129,12 +132,49 @@ def _add_segmentation_image(self, dataset):
 
         scale = get_image_scale(self.base_image_metadata, seg_meta_data)
         if dataset is not None:
-            dataset.add_object(
-                ObsSegmentationsOmeTiffWrapper(img_url=img_url, offsets_url=offsets_url,
-                                               obs_types_from_channel_names=True,
-                                               coordinate_transformations=[{"type": "scale", "scale": scale}]
-                                               )
-            )
+            if self.view_type == GEOMX_IMAGE_VIEW_TYPE:
+                dataset.add_object(
+                    ObsSegmentationsOmeTiffWrapper(img_url=img_url, offsets_url=offsets_url,
+                                                   coordination_values={"fileUid": "segmentations"},
+                                                   obs_types_from_channel_names=True,
+                                                   coordinate_transformations=[{"type": "scale", "scale": scale}]
+                                                   )
+                )
+            else:
+                dataset.add_object(
+                    ObsSegmentationsOmeTiffWrapper(img_url=img_url, offsets_url=offsets_url,
+                                                   obs_types_from_channel_names=True,
+                                                   coordinate_transformations=[{"type": "scale", "scale": scale}]
+                                                   )
+                )
+
+    def _add_aoi_rois(self, dataset):
+        segment_file_url = self._build_assets_url(self.segment_files_regex)
+        areaZarr = AnnDataWrapper(
+            adata_url=f'{segment_file_url}/aoi.zarr',
+            obs_set_paths=['obs/roi_id'],
+            obs_set_names=['ROI'],
+            coordination_values={
+                "obsType": "area"
+            },
+        )
+
+        regionZarr = AnnDataWrapper(
+            adata_url=f'{segment_file_url}/roi.zarr',
+            coordination_values={
+                "obsType": "region"
+            },
+        )
+        dataset.add_object(regionZarr)
+        dataset.add_object(areaZarr)
```

Comment:
We should stick to snake case for variable names in Python:

```suggestion
        area_zarr = AnnDataWrapper(
            adata_url=f'{segment_file_url}/aoi.zarr',
            obs_set_paths=['obs/roi_id'],
            obs_set_names=['ROI'],
            coordination_values={
                "obsType": "area"
            },
        )

        region_zarr = AnnDataWrapper(
            adata_url=f'{segment_file_url}/roi.zarr',
            coordination_values={
                "obsType": "region"
            },
        )
        dataset.add_object(region_zarr)
        dataset.add_object(area_zarr)
```